pmd-java/src/main/java/net/sourceforge/pmd/lang/java/internal/JavaLanguageProperties.java

```diff
@@ -35,11 +35,24 @@ public class JavaLanguageProperties extends JvmLanguagePropertyBundle {
                        .defaultValue(true)
                        .build();
 
+    /**
+     * @since 7.11.0
+     */
+    public static final PropertyDescriptor<Boolean> FIRST_CLASS_LOMBOK =
+        PropertyFactory.booleanProperty("lombok")
+                       .desc("Whether to consider lombok-specific things in core facilities like type inference. "
+                                 + "Disable this option if you want to analyze the AST as it would appear before the lombok pre-processing is applied. "
+                                 + "For instance, with this option enabled, variables declared with type lombok.val will have their type inferred based on the right-hand-side. "
+                                 + "With the option disabled, the variable will have type lombok.val instead. "
+                                 + "See https://github.com/pmd/pmd/issues/3119")
```

Comment:
I'll add this also to the documentation -> https://docs.pmd-code.org/latest/pmd_languages_configuration.html#java-language-properties

Reply:
Thanks. I didn't know we had that